hw/ip/spi_device/data/spi_device.hjson

```diff
@@ -656,13 +656,35 @@
           name: "addrfifo_notempty"
           desc: "Upload Address FIFO Not Empty"
         } // f: addrfifo_notempty
-        { bits: "24:16"
+      ]
+    } // R: UPLOAD_STATUS
+    { name: "UPLOAD_STATUS2"
+      desc: '''Upload module status 2 register.
+
+        This register contains payload related status. payload_depth indicates
+        the payload size (from 0 to 256 bytes).
+
+        payload_start_idx indicates the start of the 256B. This stays 0
+        usually. However, when the SPI host system issues more than 256B of
+        payload in a command, this field may not be 0. For example, if the
+        system issues 258B payload, the payload_depth is 256 (as the IP only
+        holds 256B of payload), the payload_start_idx is 2. SW should read from
+        2 to 255 then 0 and 1.
```

Comment:
Sorry for the post-merge comment: If we haven't yet, we should probably double-check our understanding of this mechanism (by doing a larger-than-a-page program operation on a real SPI flash device). It's not clear if the data bytes are assigned to an address as they are written or if the last 256 B are adjusted to begin at the address at the start of the command.

One of Winbond's data sheets suggests it's the former, and if so, SW should _not_ adjust the mapping with `payload_start_idx`, since we're already wrapping to the same addresses with the circular FIFO.